File: CSIS3000/Homework/connectedComponents/Node.java
package connectedComponents;

import java.awt.Point;
import java.util.ArrayList;

public class Node {
	public String name;
	public ArrayList<Node> connectedNodes;
	public Point nodePosition;
	public boolean isVisited;

	public Node(String name, Point nodePosition) {
		if (nodePosition == null) {
			this.nodePosition = new Point(0, 0);
		} else {
			this.nodePosition = nodePosition;
		}
		this.name = name;
		this.connectedNodes = new ArrayList<Node>();
		this.connectedNodes.add(this);
		this.isVisited = false;
	}

	public Node(String name, Point nodePosition, ArrayList<Node> connectedNodes) {
		if (nodePosition == null) {
			this.nodePosition = new Point(0, 0);
		} else {
			this.nodePosition = nodePosition;
		}

		if (connectedNodes == null) {
			this.connectedNodes = new ArrayList<Node>();
		} else {
			this.connectedNodes = connectedNodes;
		}
		this.name = name;
		this.connectedNodes.add(this);
		this.isVisited = false;
	}

	public void ConnectsTo(Node node) {
		this.connectedNodes.add(node);
		if (!node.connectedNodes.contains(this)) {
			node.connectedNodes.add(this);
		}
	}
}

File: Connected Components/Connected Components/Node.swift
import UIKit

class Node: NSObject {
	let name: String
	var connectedNodes: [Node]
	var nodePosition: CGPoint
	var isVisited: Bool
	
	init(name: String, nodePosition: CGPoint) {
		self.name = name
		self.nodePosition = nodePosition
		self.connectedNodes = [Node]()
		self.isVisited = false
	}
	
	func ConnectsTo(node: Node) {
		self.connectedNodes.append(node)
		
		if !node.connectedNodes.contains(self) {
			node.connectedNodes.append(self)
		}
	}
	
	func RemoveConnection(node: Node) {
		if let index = connectedNodes.indexOf(node) {
			connectedNodes.removeAtIndex(index)
			if let index = node.connectedNodes.indexOf(self) {
				node.connectedNodes.removeAtIndex(index)
			}
		}
	}
}
